build/stf-run-ci/tasks/setup_registry_auth.yml

```diff
@@ -90,15 +90,35 @@
   register: sto_bundle_info
   ignore_errors: true
 
+- name: Print STO bundle info
+  ansible.builtin.debug:
+    msg:
+    - "{{ sto_bundle_info }}"
+
+- name: Fail is the expected STO bundle was not correctly fetched
+  ansible.builtin.fail:
+    msg: "The requested STO bundle couldn't be retrieved from the bundle registry. Check configuration for the bundles registry and retry."
+  when: "__service_telemetry_bundle_image_path not in sto_bundle_info.stdout"
```

Comment:
Regarding verbosity, this check is the same as the assert below.

You could update the message for assert to list the failed bundle.
Something like:
```
fail_msg:
    "The requested bundle(s) ({% if __service_telemetry_bundle_image_path not in sto_bundle_info.stdout %}STO{%endif%}, {% if __smart_gateway_bundle_image_path not in sto_bundle_info.stdout %}SGO{%endif%}) couldn't be retrieved..."
```
i.e. use jinja2 templating to add in the name of the bundles(s) that weren't retrievable to the task in L98.
I'm assuming that you wanted to add these tests here so that there is a useful message to tell you which bundle was missing.

Reply:
Thanks for the review Emma. I opted for dropping the assertion altogether and just stick with the fail checks. Also, kept the debug, I think it's useful info.

Reply:
That's fair. looks good